### PR TITLE
gmp: Add native libgmp to build native gdb

### PR DESCRIPTION
**Pull request by shahab-vahedi**
GDB 11.1+ requires libgmp to be built natively [1].  Therefore, add
libgmp building to the process of building native gdb.

[1] gdb: Make GMP a required dependency for building GDB
https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=1b4ac058f7d

### CODE REVIEW

**Comment by anthony-kolesov** [comment]
I would recommend extracting the common code into arc-init.sh and calling as a function to avoid duplication (which is plentiful between uclibc and glibc files...). For example a `build_expat` function shared by all targets: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-releases/arc-init.sh#L527

File: build-glibc.sh [hunk 1]
```diff
@@ -488,6 +488,32 @@ if [ $DO_NATIVE_GDB = yes ]; then
...
+    build_dir_init gmp
+    tar xf $toolchain_build_dir/_download_tmp/$gmp_tar --strip-components=1
+
+    configure_for_arc . $triplet \
+	--prefix=/usr \
+	--sysconfdir=/etc \
+	--program-prefix= \
+	--enable-static
+    make_target building
+    make_target_ordered installing install DESTDIR=$SYSROOTDIR
```
Comment (anthony-kolesov) [inline build-glibc.sh]: Common code should be extracted into arc-init.sh , for example see a `build_expat` function shared by all targets: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-releases/arc-init.sh#L527

**Review by anthony-kolesov — CHANGES_REQUESTED** [review]

File: build-glibc.sh (hunk 1 above)
Comment (shahab-vahedi) [reply build-glibc.sh]: Done. As a bonus, I've also moved the `ncurses` part.

**Review by anthony-kolesov — APPROVED** [review]

**Comment by anthony-kolesov** [comment]
After the current GNU release it makes sense to update to the latest ncurses version and remove the hack from 2016 as it will be not needed anymore.

**Comment by shahab-vahedi** [comment]
It caught my attention too. I will tackle it at some other time. Thanks for the review!

**Pull request merged**